### PR TITLE
Suggest removing unnecessary &mut as help message

### LINKED ISSUE #68697: Suggest removing `&mut x` when `x: &mut T` and `&mut T` is expected type
An example:
```rust
struct A;

fn bar(x: &mut A) {}

fn foo(x: &mut A) {
    bar(&mut x);
}
```
results in:
```rust
error[E0596]: cannot borrow `x` as mutable, as it is not declared as mutable
 --> src/lib.rs:6:9
  |
5 | fn foo(x: &mut A) {
  |        - help: consider changing this to be mutable: `mut x`
6 |     bar(&mut x);
  |         ^^^^^^ cannot borrow as mutable
```
but we could say:
```rust
error[E0596]: cannot borrow `x` as mutable, as it is not declared as mutable
 --> src/lib.rs:6:9
  |
5 | fn foo(x: &mut A) {
6 |     bar(&mut x);
  |         ^^^^^^ cannot borrow as mutable
  |         ---- help: remove the unnecessary `&mut` here
```

### PR DESCRIPTION
**Pull request by ibraheemdev**
Closes #68697

### CODE REVIEW

**Comment by leonardo-m** [comment]
See also:
https://github.com/rust-lang/rust-clippy/issues/5546
https://github.com/rust-lang/rust/issues/83409

File: src/test/ui/nll/issue-51191.rs [hunk 1]
```diff
@@ -1,13 +1,15 @@
+#![allow(unconditional_recursion)]
+
```
Comment (wesleywiser) [inline src/test/ui/nll/issue-51191.rs:2]: Any reason we're allowing this lint? Seems like it's not really related to the rest of the changes.

Comment (ibraheemdev) [reply src/test/ui/nll/issue-51191.rs:2]: An error about an unexcepted help message showed up related to the unconditional recursion after I added the help message changes, I can revert it and add the error annotation if you want.

Comment (wesleywiser) [reply src/test/ui/nll/issue-51191.rs:2]: That would be great, thanks!

Comment (ibraheemdev) [reply src/test/ui/nll/issue-51191.rs]: Fixed.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**